tests/chart_tests/test_elasticsearch.py

```diff
@@ -530,31 +531,100 @@ def test_elasticsearch_curator_cronjob_defaults(self, kube_version):
             "sleep 5; /usr/bin/curator --config /etc/config/config.yml /etc/config/action_file.yml; exit_code=$?; wget --timeout=5 -O- --post-data='not=used' http://127.0.0.1:15020/quitquitquit; exit $exit_code;"
         ]
         assert c_by_name["curator"]["securityContext"] == {}
 
     def test_elasticsearch_curator_cronjob_overrides(self, kube_version):
-        """Test ElasticSearch Curator cron job with defaults"""
+        """Test ElasticSearch Curator cron job with nodeSelector, affinity, tolerations and config defaults"""
+        values = {
+            "elasticsearch": {
+                "curator": {
+                    "schedule": "0 45 * * *",
+                    "securityContext": {"runAsNonRoot": True},
+                }
+            },
+            "global": {
+                "platformNodePool": global_platform_node_pool_config,
+            },
+        }
         docs = render_chart(
             kube_version=kube_version,
-            values={
-                "elasticsearch": {
-                    "curator": {
-                        "schedule": "0 45 * * *",
-                        "securityContext": {"runAsNonRoot": True},
+            values=values,
+            show_only=["charts/elasticsearch/templates/curator/es-curator-cronjob.yaml"],
+        )
+        assert len(docs) == 1
+        c_by_name = get_cronjob_containerspec_by_name(docs[0])
+        assert docs[0]["kind"] == "CronJob"
+        assert docs[0]["metadata"]["name"] == "release-name-elasticsearch-curator"
+        assert docs[0]["spec"]["schedule"] == "0 45 * * *"
+        assert docs[0]["spec"]["jobTemplate"]["spec"]["template"]["spec"]["nodeSelector"]["role"] == "astro"
+        assert len(docs[0]["spec"]["jobTemplate"]["spec"]["template"]["spec"]["affinity"]) == 1
+        assert len(docs[0]["spec"]["jobTemplate"]["spec"]["template"]["spec"]["tolerations"]) > 0
+        assert (
+            docs[0]["spec"]["jobTemplate"]["spec"]["template"]["spec"]["tolerations"]
+            == values["global"]["platformNodePool"]["tolerations"]
+        )
+        assert c_by_name["curator"]["command"] == ["/bin/sh", "-c"]
+        assert c_by_name["curator"]["args"] == [
+            "sleep 5; /usr/bin/curator --config /etc/config/config.yml /etc/config/action_file.yml; exit_code=$?; wget --timeout=5 -O- --post-data='not=used' http://127.0.0.1:15020/quitquitquit; exit $exit_code;"
+        ]
+        assert c_by_name["curator"]["securityContext"] == {"runAsNonRoot": True}
+
+    def test_elasticsearch_curator_cronjob_subchart_overrides(self, kube_version):
+        """Test ElasticSearch Curator cron job with nodeSelector, affinity, tolerations and config defaults"""
+        values = {
+            "elasticsearch": {
+                "curator": {
+                    "schedule": "0 45 * * *",
+                    "securityContext": {"runAsNonRoot": True},
+                },
+                "nodeSelector": {"role": "astroelasticsearch"},
```

Comment:
If we're going to put global_platform_node_pool_config in __init__.py, shouldn't we deepcopy that and update the fields we need to change?